### PR TITLE
Apply `black` to `astropy.time`

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Apply `black` to `astropy.time` according to the process laid out by [APE 20](https://github.com/astropy/astropy-APEs/blob/main/APE20.rst).

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label. Codestyle issues can be fixed by the [bot](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit).
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Review by taldcroft — APPROVED** [review]

**Review by mhvk — APPROVED** [review]
Looks OK to me too, but I couldn't resist replacing most `.format` in exceptions with f-strings.

File: astropy/time/core.py [hunk 1]
```diff
@@ -650,7 +647,7 @@ def format(self):
     def format(self, format):
         """Set time format"""
         if format not in self.FORMATS:
-            raise ValueError(f'format must be one of {list(self.FORMATS)}')
+            raise ValueError(f'format must be one of {sorted(self.FORMATS)}')
```
Comment (taldcroft) [inline astropy/time/core.py:650]: This is an API change and needs to be reverted.
```
In [4]: sorted(TIME_FORMATS) == list(TIME_FORMATS)
Out[4]: False
```

File: astropy/time/core.py [hunk 2]
```diff
@@ -968,7 +963,7 @@ def to_value(self, format, subfmt='*'):
         # TODO: add a precision argument (but ensure it is keyword argument
         # only, to make life easier for TimeDelta.to_value()).
         if format not in self.FORMATS:
-            raise ValueError(f'format must be one of {list(self.FORMATS)}')
+            raise ValueError(f'format must be one of {sorted(self.FORMATS)}')
```
Comment (taldcroft) [inline astropy/time/core.py:966]: API change, needs to be reverted.

File: astropy/time/core.py [hunk 3]
```diff
@@ -1329,7 +1320,7 @@ def _apply(self, method, *args, format=None, cls=None, **kwargs):
         # in the copy.  If the format is unchanged this process is lightweight
         # and does not create any new arrays.
         if new_format not in tm.FORMATS:
-            raise ValueError(f'format must be one of {list(tm.FORMATS)}')
+            raise ValueError(f'format must be one of {sorted(tm.FORMATS)}')
```
Comment (taldcroft) [inline astropy/time/core.py:1323]: API change, revert.

File: astropy/time/core.py [hunk 4]
```diff
@@ -1600,7 +1591,7 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
         if np.any(divisor == 0):
             raise ValueError(
                 'Mean over zero elements is not supported as it would give an undefined'
-                ' time;see issue https://github.com/astropy/astropy/issues/6509'
+                ' time; see issue https://github.com/astropy/astropy/issues/6509'
```
Comment (taldcroft) [inline astropy/time/core.py:1594]: API change, revert.

File: astropy/time/core.py [hunk 5]
```diff
@@ -1683,9 +1673,8 @@ def _match_shape(self, val):
                 val = np.broadcast_to(val, self.shape, subok=True)
             except Exception:
                 raise ValueError(
-                    'Attribute shape must match or be '
-                    'broadcastable to that of Time object. '
-                    'Typically, give either a single value or '
+                    'Attribute shape must match or be broadcastable to that of '
+                    'the Time object. Typically, give either a single value or '
```
Comment (taldcroft) [inline astropy/time/core.py:1677]: API change (in wording), revert.

File: astropy/time/core.py [hunk 6]
```diff
@@ -3128,9 +3103,7 @@ def to_value(self, *args, **kwargs):
             except ValueError as exc:
                 raise ValueError(
                     "first argument is not one of the known "
-                    "formats ({}) and failed to parse as a unit.".format(
-                        list(self.FORMATS)
-                    )
+                    f"formats ({sorted(self.FORMATS)}) and failed to parse as a unit."
```
Comment (taldcroft) [inline astropy/time/core.py:3106]: API change, revert to `list(self.FORMATS)`.

**Review by taldcroft — COMMENTED** [review]
@mhvk - these commits introduce changes in exception messages that count as API changes since application code could be checking those messages.

**Comment by mhvk** [comment]
OK, you're right, I shouldn't have tried to introduce improvements in PRs that are just meant not to change anything. Now reverted (the ones you noted and one additional one). Sorry for the extra work!

**Pull request merged**